gips/0053-enabling-substreams-based-subgraphs.md

```diff
@@ -0,0 +1,138 @@
+
+A substreams package is a collection of interdependent modules, which combine to extract and transform blockchain events in a highly configurable and parallelisable way, producing a streaming output.
+
+This GIP describes the simplest possible integration of Substreams with Subgraphs, to take that final streaming output and make it persistent and queryable.
+
+# Motivation
+
+As Substreams become first-class citizens with The Graph Network, the StreamingFast team believes that Substreams software has been battle tested enough to be deemed fully production ready with The Graph Network. To ensure that Indexers will index substreams-back subgraphs, the GIP proposes to add Oracle support for dataSource.kind =="substreams" to the Feature Matrix shared below. 
+
+This would make an important moment where the performance promises we have made in the last 2 years come to fruition. With a very important indexing-time performance boost, as well as an important injection-time performance boost.
+
+Adding Substream support to Graph Node is the fastest way to bring the performance and composability benefits of substreams to The Graph Network.
+
+# High Level Description
+
+With the launch of Uniswap v3, lots of efforts have been put in stabilizing the engine. In doing so, multiple indeterminism issues were resolved. StreamingFast has done extensive testing for determinism, have validated substreams-graph-load (the high-speed injector) and graph-node loading PoI parity, as well as the gentle hand-off to graph-node with continued parity.
+
+StreamingFast’s testing also showed convergence of POIs during live segments of the chain, even if hit by potential reorgs (this uses the standard support in graph-node so shouldn’t introduce any new risks).
+
+The release Release v1.4.4 · streamingfast/firehose-ethereum · GitHub 1 is required for deterministic execution.
+
+# Detailed Specification
+
+```typescript
+specVersion: 0.0.X
+schema:
+  file: ./src/schema.graphql
+dataSources:
+  - kind: substreams
+    name: Uniswap
+    network: mainnet
+    source:
+      package:
+        moduleName: output
+        file:
+          /: /ipfs/QmbHnhUFZa6qqqRyubUYhXntox1TCBxqryaBM1iNGqVJzT
+          # This IPFs path would be generated from a local path at deploy time
+    mapping:
+      kind: substreams/graph-entities
+      apiVersion: 0.0.X
+```   
+
+This introduces a new `substreams` “kind” of dataSource. This dataSource is identified on the `subgraph.yaml` by a name and a filepath. This filepath will point locally to a Substreams package, which will be uploaded to IPFS on deployment.
+
+> Subgraphs with a substreams dataSource can only have that single dataSource.
+
+#### Graph Node configuration
+In order to support substream-based subgraphs, Graph Node will need to be Substream aware. An additional substream endpoint will be required for each network which Graph Node will want to support
```

Comment:
This GIP was written before the implementation, should the language be updated to use "past tense" more than "future tense"?